quic/s2n-quic-core/src/inet/ipv4.rs

```diff
@@ -305,6 +314,330 @@ impl From<IpV4Address> for [u8; IPV4_LEN] {
     }
 }
 
+//= https://www.rfc-editor.org/rfc/rfc791.html#section-3.1
+//#  A summary of the contents of the internet header follows:
+//#
+//#
+//#    0                   1                   2                   3
+//#    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//#   |Version|  IHL  |Type of Service|          Total Length         |
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//#   |         Identification        |Flags|      Fragment Offset    |
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//#   |  Time to Live |    Protocol   |         Header Checksum       |
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//#   |                       Source Address                          |
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//#   |                    Destination Address                        |
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+//#   |                    Options                    |    Padding    |
+//#   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+define_inet_type!(
```

Comment:
Are the `Options` and `Padding` bits not necessary to represent in the Header structure? Might be nice to include a note about why not.

Reply:
Yeah the options is variable length depending on the header len field. I can include a note.